websites/M/MyOnlineRadio/handleStation.ts

```diff
@@ -0,0 +1,139 @@
+	switch (action) {
+		case "playlist":
+			{
+				searchToPres(presenceData, station);
+				try {
+					const yPath = new URL(
+						document.querySelector<HTMLIFrameElement>(
+							'[src*="//www.youtube.com/embed/"]'
+						)?.src
+					).pathname.split("/")[2];
+					if (yPath) {
+						const watching = {
+							url: `https://www.youtube.com/watch?v=${yPath}`,
+							title:
+								document
+									.querySelector(`[data-youtube="${yPath}"]`)
+									?.querySelector(".txtsong")?.textContent || "",
+						};
+						presenceData.details = `Watching ${watching.title}`;
+						presenceData.state = `From ${station}`;
+						presenceData.buttons = [
+							{ label: "View on YouTube", url: watching.url },
+						];
+					}
+				} catch (err) {
+					/* empty */
+				}
+			}
+			break;
+
+		case "archive":
+			{
+				searchToPres(presenceData, station);
+				const audioPlayer = document.querySelector('[title="Pause"]');
+				if (audioPlayer) {
+					presenceData.details = `From ${station}`;
+
+					// no there is not a better way, all the tags are the same
+					const audioName =
+						audioPlayer.parentElement.parentElement.parentElement.parentElement.querySelector(
+							"h4"
+						)?.innerText;
+					presenceData.state = `Listening to ${audioName}`;
+					const startTS = audioPlayer.parentElement.querySelector(
+							".audioplayer-time-current"
+						)?.textContent,
+						endTS = audioPlayer.parentElement.querySelector(
+							".audioplayer-time-duration"
+						)?.textContent;
+
+					if (startTS) presenceData.startTimestamp = tDiffInMS(startTS);
+					if (startTS && endTS)
+						presenceData.endTimestamp = tDiffInMS(startTS, endTS);
+				}
+			}
+			break;
+
+		case "frequency":
+			{
+				presenceData.details = station;
+				presenceData.state = "Viewing Frequency Map";
+				if (presenceData.buttons) presence.clearActivity();
+			}
+			break;
+
+		case "schedule":
+			{
+				presenceData.details = `Viewing Schedule for ${station}`;
+				delete presenceData.state;
+				searchToPres(presenceData, station);
+			}
+			break;
+
+		case "news":
+			{
+				presenceData.details = `Viewing News for ${station}`;
+				delete presenceData.state;
+			}
+			break;
+
+		default: {
+			presenceData.details = `Browsing ${station}`;
+			presenceData.buttons = [
+				{ label: "View Station", url: document.location.href },
+			];
+
+			const play = document.querySelector('[data-original-title="Pause"]');
+			if (play) {
+				presenceData.state = `Listening to ${
+					document.querySelector(".actSong")?.textContent
+				}`;
+				presenceData.startTimestamp = tDiffInMS(
+					document.querySelector(".ctime").textContent
+				);
+				presenceData.largeImageText = document
+					.querySelector(".server-conts")
+					?.querySelector(".active > .info")?.textContent;
+
+				// check if they're recording
+				if (document.querySelector(".live")) {
+					presenceData.smallImageKey = "https://i.imgur.com/4xhphHG.png";
```

Comment:
Image returning 404 not found, please try uploading to a different image website, imgur seems to be unstable recently.